### PR TITLE
WIP: Implement getUserProfile

### PR DESCRIPTION
**Pull request by gunnar-sendirad**
We weren't told what to return from this function, so it seems to make sense to just return the object we got from the client call.

That client has more functions. We had talked about only implementing the read ones, that is, the ones that use HTTP GET. Those are:

- userProfileControllerFindOneByNationalId
- userTokenControllerFindOneByNationalId
- userTokenControllerGetDeviceTokens

This commit implements the first one. We didn't (yet) do the second two because the module this gets used in is UserProfileModule.  As far as I can tell, that module is configured in SharedDataProviders, where it appears as an instance of ApplicationTemplateAPIAction. That interface has a single field called apiModuleAction, and our original module is only configured with getUserProfile as that action. So... we're done?

### CODE REVIEW

File: libs/api/domains/user-profile/src/index.ts [hunk 1]
```diff
@@ -1,2 +1,3 @@
 export * from './lib/userProfile.module'
 export * from './lib/userProfile.resolver'
+export { IslykillService } from './lib/islykill.service'
```
Comment (obmagnusson) [inline libs/api/domains/user-profile/src/index.ts:3]: Remove

File: libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.module.ts [hunk 2]
```diff
@@ -11,14 +11,24 @@ import { BaseTemplateAPIModuleConfig } from '../../../../types'
 import { UserProfileService } from './user-profile.service'
 
 import { Configuration, UserProfileApi } from '@island.is/clients/user-profile'
+import { IslykillService } from '@island.is/api/domains/user-profile'
```
Comment (obmagnusson) [inline libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.module.ts:14]: We don't want to import from the API Layer. Should be purely using the Client libs.

File: libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.module.ts [hunk 3]
```diff
@@ -11,14 +11,24 @@ import { BaseTemplateAPIModuleConfig } from '../../../../types'
...
+        SharedTemplateAPIModule.register(config),
+        IslykillApiModule.register({
+          cert: config.islykill.cert,
+          passphrase: config.islykill.passphrase,
+          basePath: config.islykill.basePath,
+        }),
+      ],
       providers: [
         UserProfileService,
+        IslykillService,
```
Comment (obmagnusson) [inline libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.module.ts:31]: Should just be able to import the Islykillmodule and inject the service. Add it to export within the module if not exported.

File: libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.service.ts [hunk 4]
```diff
@@ -1,45 +1,135 @@
...
-import { ProblemError } from '@island.is/nest/problem'
-import { ProblemType } from '@island.is/shared/problem'
-import { isRunningOnEnvironment } from '@island.is/shared/utils'
+import { UserProfileApi, UserProfile } from '@island.is/clients/user-profile'
+import { IslykillService } from '@island.is/api/domains/user-profile'
+import { logger } from '@island.is/logging'
 import { Injectable } from '@nestjs/common'
 import { SharedTemplateApiService } from '../..'
 import { TemplateApiModuleActionProps } from '../../../../types'
+import { ApolloError } from 'apollo-server-express'
```
Comment (obmagnusson) [inline libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.service.ts:8]: I think we can safely remove the Appollo Gql dependencies in here.

File: libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.service.ts [hunk 5]
```diff
@@ -1,45 +1,135 @@
...
+        const outOfDateEmailMobile = (emptyMail || emptyMobile) && diffOutOfDate
 
-      return {
-        email: profile.email,
-        emailVerified: profile.emailVerified,
-        mobilePhoneNumber: profile.mobilePhoneNumber,
-        mobilePhoneNumberVerified: profile.mobilePhoneNumberVerified,
-      }
-    } catch (e) {
-      if (isRunningOnEnvironment('local')) {
```
Comment (obmagnusson) [inline libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.service.ts:36]: We dont want everyone working on the application system having to set up and running the userprofile service and or islykill.

File: libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.service.ts [hunk 6]
```diff
@@ -1,45 +1,135 @@
...
   }
 
-  async getUserProfile({ auth }: TemplateApiModuleActionProps) {
-    try {
-      //TODO finish this
-      const profile = await this.userProfileApiWithAuth(
-        auth,
-      ).userProfileControllerFindOneByNationalId({
-        nationalId: auth.nationalId,
+  // eslint-disable-next-line
```
Comment (obmagnusson) [inline libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.service.ts:37]: Why is this needed?

**Review by obmagnusson — CHANGES_REQUESTED** [review]

File: libs/api/domains/user-profile/src/index.ts (hunk 1 above)
Comment (gunnar-sendirad) [reply libs/api/domains/user-profile/src/index.ts]: Gone

File: libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.module.ts (hunk 2 above)
Comment (gunnar-sendirad) [reply libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.module.ts]: Done

File: libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.service.ts (hunk 4 above)
Comment (gunnar-sendirad) [reply libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.service.ts]: Done

File: libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.module.ts (hunk 3 above)
Comment (gunnar-sendirad) [reply libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.module.ts]: Done.

File: libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.service.ts [hunk 7]
```diff
@@ -1,45 +1,135 @@
...
+         * Even if userProfileApiWithAuth does not exist.
+         * Islykill data might exist for the user, so we need to get that, with default values in the userprofile data.
+         */
+        return this.getIslykillProfile(auth)
+      }
+      handleError(error)
+    })
   }
-}
+}
```
[the remark this answers is not in the document]
Comment (gunnar-sendirad) [reply libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.service.ts]: \n

File: libs/api/domains/user-profile/src/index.ts [hunk 8]
```diff
@@ -1,2 +1,2 @@
 export * from './lib/userProfile.module'
-export * from './lib/userProfile.resolver'
+export * from './lib/userProfile.resolver'
```
Comment (obmagnusson) [inline libs/api/domains/user-profile/src/index.ts:2]: New line

**Comment by obmagnusson** [comment]
Implemented within #7280
